packages/react-components/src/components/VideoGallery/LargeGalleryLayout.tsx

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { useMemo, useRef, useState } from 'react';
+import { isNarrowWidth } from '../utils/responsive';
+/* @conditional-compile-remove(gallery-layouts) */
+import { isShortHeight } from '../utils/responsive';
+import { LayoutProps } from './Layout';
+import { OverflowGallery } from './OverflowGallery';
+import { GridLayout } from '../GridLayout';
+import { Stack } from '@fluentui/react';
+import { useOrganizedParticipants } from './utils/videoGalleryLayoutUtils';
+import { rootLayoutStyle } from './styles/DefaultLayout.styles';
+import { videoGalleryLayoutGap } from './styles/Layout.styles';
+/* @conditional-compile-remove(gallery-layouts) */
+import { VERTICAL_GALLERY_TILE_SIZE_REM } from './styles/VideoGalleryResponsiveVerticalGallery.styles';
+
+/**
+ * Props for {@link LargeGalleryLayout}.
+ *
+ * @private
+ */
+export type LargeGalleryProps = LayoutProps;
+
+/**
+ * VideoGallery Layout for when user is in a large meeting and wants to see more participants
+ *
+ * Caps the number of tiles that a participant can see in the grid to 49, Video and Audio.
+ *
+ * @private
+ */
+export const LargeGalleryLayout = (props: LargeGalleryProps): JSX.Element => {
+  const {
+    remoteParticipants = [],
+    localParticipant,
+    dominantSpeakers,
+    localVideoComponent,
+    screenShareComponent,
+    onRenderRemoteParticipant,
+    styles,
+    maxRemoteVideoStreams,
+    parentWidth,
+    /* @conditional-compile-remove(gallery-layouts) */ parentHeight,
+    pinnedParticipantUserIds = [],
+    /* @conditional-compile-remove(vertical-gallery) */ overflowGalleryPosition = 'HorizontalBottom'
+  } = props;
+
+  const isNarrow = parentWidth ? isNarrowWidth(parentWidth) : false;
+  /* @conditional-compile-remove(gallery-layouts) */
+  const isShort = parentHeight ? isShortHeight(parentHeight) : false;
+
+  const maxStreamsTrampoline = (): number => {
+    /* @conditional-compile-remove(gallery-layouts) */
+    return parentWidth && parentHeight
+      ? calculateMaxTilesInLargeGrid(parentWidth, parentHeight)
+      : maxRemoteVideoStreams;
+    return maxRemoteVideoStreams;
+  };
+
+  // This is for tracking the number of children in the first page of overflow gallery.
+  // This number will be used for the maxOverflowGalleryDominantSpeakers when organizing the remote participants.
+  // We need to add the local participant to the pinned participant count so we are placing the speakers correctly.
+  const childrenPerPage = useRef(4);
```

Comment:
where did `4` come frome? use defined constants with names for arbitrary numbers. 
Why 4 and not 5 like horiztonal gallery?
![image](https://github.com/Azure/communication-ui-library/assets/2684369/14bd3811-a95f-4816-bede-bf6daf8872f2)